Fix Type Annotation in pandas.compat.pickle_compat

Part of #25882 

Current errors are as follows:

>mypy pandas/compat/pickle_compat.py

    pandas\compat\pickle_compat.py:141: error: Name 'pkl._Unpickler' is not defined

--- a/mypy.ini
+++ b/mypy.ini
@@ -11,9 +11,6 @@ ignore_errors=True
 [mypy-pandas.compat.numpy.function]
 ignore_errors=True
 
-[mypy-pandas.compat.pickle_compat]
-ignore_errors=True
-
 [mypy-pandas.core.accessor]
 ignore_errors=True
 

--- a/pandas/compat/pickle_compat.py
+++ b/pandas/compat/pickle_compat.py
@@ -138,7 +138,7 @@ _class_locations_map = {
 # our Unpickler sub-class to override methods and some dispatcher
 # functions for compat
 
-class Unpickler(pkl._Unpickler):
+class Unpickler(pkl._Unpickler):  # type: ignore
 
     def find_class(self, module, name):
         # override superclass

